pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -486,8 +486,9 @@ func (c *Controller) deleteSecret(secretKey string) {
 				clusterID, secretKey, err)
 		}
 		close(cluster.Stop)
-		delete(c.cs.remoteClusters, secretKey)
+		delete(c.cs.remoteClusters[secretKey], clusterID)
```

Comment:
Maybe donot need this at all, not sure which is best for go gc

Reply:
yes, maybe just need delete from map using `secretKey` finally after range